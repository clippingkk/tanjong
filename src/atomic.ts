import { ProfileQuery } from "./schema/generated"
import { atomWithMMKV } from "./utils/jotai"

export const tokenAtom = atomWithMMKV<string | null>('me:token', null)
export const uidAtom = atomWithMMKV<number | null>('me:uid', null)
export const profileAtom = atomWithMMKV<ProfileQuery['me'] | null>('me:profile', null)