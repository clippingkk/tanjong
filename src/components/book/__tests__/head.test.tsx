import 'react-native'
import React from 'react'
import { act, render, screen } from '@testing-library/react-native'
import { NativeBaseProvider } from 'native-base'
import { NavigationContainer } from '@react-navigation/native'
import { Provider } from 'jotai'
import { uidAtom } from '../../../atomic'
import { createNativeStackNavigator } from '@react-navigation/native-stack'
import BookHead from '../head'
import { GluestackUIProvider } from '@gluestack-ui/themed'
import { config } from '@gluestack-ui/config'

test('head book will correctly rendering', async () => {
  const now = new Date()
  const st = createNativeStackNavigator()
  function BH() {
    return (
      <BookHead
        book={{
          id: 1,
          title: 'happy chinese year',
          author: 'annatarhe',
          summary: 'summmary',
          rating: 3,
          pubdate: now.toString(),
          totalPages: 333,
          originTitle: 'hhhhhh',
          image: 'x',
          doubanId: 1,
          press: 'sssssssssg',
          url: 'https://h.c.n.y/hcny',
          isbn: 'xxxx',
          tags: ['h', 'e', 'a'],
          authorIntro: 'z',
          createdAt: now.toString(),
          updatedAt: now.toString()
        }}
      />
    )
  }
  //ScreenComponentType
  // initialValues={[[uidAtom, 1]]}
  render(
    <NativeBaseProvider initialWindowMetrics={{
      frame: { x: 0, y: 0, width: 0, height: 0 },
      insets: { top: 0, left: 0, right: 0, bottom: 0 },
    }}>
      <GluestackUIProvider config={config}>
        <Provider >
          <NavigationContainer>
            <st.Navigator>
              <st.Screen
                component={BH}
                name='/h'
              />
            </st.Navigator>
          </NavigationContainer>
        </Provider>
      </GluestackUIProvider>
    </NativeBaseProvider>
  )
  // make sure every thing rendered
  await act(() => { })
  expect(screen).toMatchSnapshot()
})
