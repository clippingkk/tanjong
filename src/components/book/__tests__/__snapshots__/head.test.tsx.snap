// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`head book will correctly rendering 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      [
        {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <RNSScreenStack
      style={
        {
          "flex": 1,
        }
      }
    >
      <RNSScreen
        collapsable={false}
        gestureResponseDistance={
          {
            "bottom": -1,
            "end": -1,
            "start": -1,
            "top": -1,
          }
        }
        nativeBackButtonDismissalEnabled={false}
        onAppear={[Function]}
        onDisappear={[Function]}
        onDismissed={[Function]}
        onGestureCancel={[Function]}
        onHeaderBackButtonClicked={[Function]}
        onNativeDismissCancelled={[Function]}
        onTransitionProgress={[Function]}
        onWillDisappear={[Function]}
        replaceAnimation="push"
        sheetAllowedDetents="large"
        sheetCornerRadius={-1}
        sheetExpandsWhenScrolledToEdge={true}
        sheetGrabberVisible={false}
        sheetLargestUndimmedDetent="all"
        stackPresentation="push"
        style={
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        swipeDirection="horizontal"
      >
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="auto"
          style={
            {
              "flex": 1,
              "flexDirection": "column-reverse",
            }
          }
        >
          <View
            collapsable={false}
            style={
              [
                {
                  "flex": 1,
                },
                {
                  "backgroundColor": "rgb(242, 242, 242)",
                },
                undefined,
              ]
            }
          >
            <View
              dataSet={{}}
              style={
                {
                  "marginBottom": -31,
                }
              }
            >
              <View
                dataSet={{}}
                style={
                  {
                    "height": 344,
                  }
                }
              >
                <View
                  style={
                    [
                      {
                        "backgroundColor": "transparent",
                      },
                      {
                        "bottom": 0,
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      },
                    ]
                  }
                >
                  <Image
                    accessibilityRole="image"
                    accessible={true}
                    onError={[Function]}
                    onLoad={[Function]}
                    resizeMode="cover"
                    source={
                      {
                        "uri": "x",
                      }
                    }
                    style={
                      [
                        {
                          "bottom": 0,
                          "left": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        },
                        {
                          "opacity": 0,
                        },
                        undefined,
                      ]
                    }
                  />
                </View>
                <BlurView
                  blurAmount={30}
                  blurType="light"
                  reducedTransparencyFallbackColor="white"
                  style={
                    [
                      {
                        "backgroundColor": "transparent",
                      },
                      {
                        "bottom": 0,
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      },
                    ]
                  }
                />
                <View
                  dataSet={{}}
                  style={
                    {
                      "alignItems": "flex-start",
                      "flexDirection": "row",
                      "justifyContent": "space-around",
                      "marginLeft": 8,
                      "marginRight": 8,
                      "marginTop": 94,
                    }
                  }
                >
                  <View
                    style={
                      [
                        {
                          "backgroundColor": "transparent",
                        },
                        [
                          {
                            "aspectRatio": 1.7777777777777777,
                            "borderRadius": 8,
                            "height": 150,
                          },
                          {
                            "shadowColor": "#000",
                            "shadowOffset": {
                              "height": 2,
                              "width": 0,
                            },
                            "shadowRadius": 3.84,
                          },
                        ],
                      ]
                    }
                  >
                    <Image
                      accessibilityRole="image"
                      accessible={true}
                      onError={[Function]}
                      onLoad={[Function]}
                      resizeMode="cover"
                      source={
                        {
                          "uri": "x",
                        }
                      }
                      style={
                        [
                          {
                            "bottom": 0,
                            "left": 0,
                            "position": "absolute",
                            "right": 0,
                            "top": 0,
                          },
                          {
                            "opacity": 0,
                          },
                          undefined,
                        ]
                      }
                    />
                  </View>
                  <View
                    dataSet={{}}
                    style={
                      {
                        "maxWidth": 240,
                      }
                    }
                  >
                    <Text
                      dataSet={{}}
                      selectable={true}
                      style={
                        {
                          "backgroundColor": undefined,
                          "color": "#171717",
                          "fontFamily": "LXGWWenKai-Regular",
                          "fontSize": 20,
                          "fontStyle": "normal",
                          "fontWeight": "400",
                          "letterSpacing": 0,
                          "lineHeight": 30,
                          "textDecorationLine": undefined,
                        }
                      }
                    >
                      happy chinese year
                    </Text>
                    <Text
                      dataSet={{}}
                      selectable={true}
                      style={
                        {
                          "backgroundColor": undefined,
                          "color": "#171717",
                          "fontFamily": "LXGWWenKai-Regular",
                          "fontSize": 14,
                          "fontStyle": "normal",
                          "fontWeight": "400",
                          "letterSpacing": 0,
                          "lineHeight": 21,
                          "textDecorationLine": undefined,
                        }
                      }
                    >
                      annatarhe
                    </Text>
                    <View
                      aria-orientation="horizontal"
                      dataSet={{}}
                      style={
                        {
                          "backgroundColor": "#d4d4d4",
                          "height": 1,
                          "marginBottom": 8,
                          "marginTop": 8,
                          "width": "100%",
                        }
                      }
                      thickness="1"
                    />
                    <Text
                      dataSet={{}}
                      numberOfLines={4}
                      style={
                        {
                          "backgroundColor": undefined,
                          "color": "#171717",
                          "fontFamily": "LXGWWenKai-Regular",
                          "fontSize": 14,
                          "fontStyle": "normal",
                          "fontWeight": "400",
                          "letterSpacing": 0,
                          "lineHeight": 21,
                          "textDecorationLine": undefined,
                        }
                      }
                    >
                      summmary
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
        <RNSScreenStackHeaderConfig
          backButtonInCustomView={false}
          backTitleVisible={true}
          backgroundColor="rgb(255, 255, 255)"
          color="rgb(0, 122, 255)"
          direction="ltr"
          disableBackButtonMenu={false}
          hidden={false}
          hideBackButton={false}
          largeTitleHideShadow={false}
          title="/h"
          titleColor="rgb(28, 28, 30)"
          topInsetEnabled={false}
          translucent={false}
        />
      </RNSScreen>
    </RNSScreenStack>
  </View>
</RNCSafeAreaProvider>
`;
